indra/statements.py

```diff
@@ -95,23 +131,13 @@ def refinement_of(self, other, entity_hierarchy, mod_hierarchy):
         # MODIFICATIONS
         # Similar to the above, we check that self has all of the modifications
         # of other.
-        assert len(self.mods) == len(self.mod_sites), \
-               "Mods and mod_sites must match."
-        assert len(other.mods) == len(other.mod_sites), \
-               "Mods and mod_sites must match."
         # Make sure they have the same modifications
         for other_mod_ix in range(len(other.mods)):
```

Comment:
We don't need indexing for these anymore because we don't have two parallel lists.